prestoadmin/node.py

```diff
@@ -40,21 +41,29 @@ def get_conf(self):
                 _LOGGER.debug('%s configuration for %s not found.  '
                               'Default configuration will be deployed',
                               type(self).__name__, name)
+                conf_value = self.default_config(name)
+                conf[name] = conf_value
+                file_path = os.path.join(self._get_conf_dir(), name)
+                config.write_conf_to_file(conf_value, file_path)
 
-        defaults = self.build_defaults()
-        config.fill_defaults(conf, defaults)
         self.validate(conf)
         return conf
 
     @abstractmethod
     def _get_conf_dir(self):
-        pass
+        raise NotImplementedError()
```

Comment:
not needed.  I'm switching it back.  I think I switched it because pycharm gives a type check warning, but it's not a correct warning..